launcher/ui/pages/instance/ModFolderPage.cpp

```diff
@@ -145,10 +146,17 @@ void ModFolderPage::downloadMods()
         QMessageBox::critical(this, tr("Error"), tr("Please install a mod loader first!"));
         return;
     }
-    auto mdownload = new ResourceDownload::ModDownloadDialog(this, m_model, m_instance);
-    mdownload->setAttribute(Qt::WA_DeleteOnClose);
-    connect(this, &QObject::destroyed, mdownload, &QDialog::close);
-    if (mdownload->exec()) {
+    
+    m_downloadDialog = new ResourceDownload::ModDownloadDialog(this, m_model, m_instance);
+    m_downloadDialog->setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
I would still remove this and use deleteLater

Reply:
or even a qt smart pointer

Reply:
I tested both, either will results in a use after free. Guarded `QPointer` is the best solution avalible.  your right that the attribute isn't needed anymore though.